vmm/src/cpu.rs

```diff
@@ -374,42 +315,11 @@ impl Vcpu {
         match self.vcpu.run() {
             Ok(run) => match run {
                 #[cfg(target_arch = "x86_64")]
-                VmExit::IoIn(addr, data) => {
-                    if let Err(e) = self.io_bus.read(u64::from(addr), data) {
-                        if let vm_device::BusError::MissingAddressRange = e {
-                            warn!("Guest PIO read to unregistered address 0x{:x}", addr);
-                        }
-                    }
-                    Ok(true)
-                }
+                VmExit::IoIn(_, _) => Ok(true),
                 #[cfg(target_arch = "x86_64")]
-                VmExit::IoOut(addr, data) => {
-                    if addr == DEBUG_IOPORT && data.len() == 1 {
-                        self.log_debug_ioport(data[0]);
-                    }
-                    if let Err(e) = self.io_bus.write(u64::from(addr), data) {
-                        if let vm_device::BusError::MissingAddressRange = e {
-                            warn!("Guest PIO write to unregistered address 0x{:x}", addr);
-                        }
-                    }
-                    Ok(true)
-                }
-                VmExit::MmioRead(addr, data) => {
-                    if let Err(e) = self.mmio_bus.read(addr as u64, data) {
-                        if let vm_device::BusError::MissingAddressRange = e {
-                            warn!("Guest MMIO read to unregistered address 0x{:x}", addr);
-                        }
-                    }
-                    Ok(true)
-                }
-                VmExit::MmioWrite(addr, data) => {
-                    if let Err(e) = self.mmio_bus.write(addr as u64, data) {
-                        if let vm_device::BusError::MissingAddressRange = e {
-                            warn!("Guest MMIO write to unregistered address 0x{:x}", addr);
-                        }
-                    }
-                    Ok(true)
-                }
+                VmExit::IoOut(_, _) => Ok(true),
+                VmExit::MmioRead(_, _) => Ok(true),
+                VmExit::MmioWrite(_, _) => Ok(true),
```

Comment:
Since we're setting  `VmmOps`, we should never get those `VmExit` back from the hypervisor crate. We should only get `VmExit::Ignore` for PIO and MMIO. So I think here we want to send a warning at least.

Reply:
Well you assume the `hypervisor/src/kvm/mod.rs` implementation will return `VmExit::Ignore`. Is that what we want? I'm totally fine with changing the code, but my point is that the KVM implementation should return `VmExit::Ignore`.

Reply:
Nevermind... The KVM code already does that!

Reply:
Done! Please take a look.